File: javascript/242. Valid Anagram.js
function validAnagram(s1, s2) {
    countDict = {};
    for (char of s1){
        countDict[char] = (countDict[char] || 0) + 1;
    };
    
    for (char of s2){
        if(!(char in countDict))return false
        else{
            countDict[char]--
            if (countDict[char]==0) delete  countDict[char]
        }
    };
    return Object.keys(countDict).length ===0
};

let s = "anagram";
let t = "nagaram";
validAnagram(s, t)

File: 1.array/1.easy/242. Valid Anagram.py
def validAnagram(first, second):
    
    if len(first) != len(second) : return False

    count = {}

    for i in first:
        if i not in count:
            count[i] = 1
        else:
            count[i] += 1

    
    for x in second:
        if x not in count:
            return False
        else:
           count[x] -= 1
           if count[x] == 0:count.pop(x)

    return not count

s = "anagram"
t = "nagaram"
validAnagram(s, t)
